Switch to using the v2 resolver in most workspaces

tgross35 opened the pull request:
Pinning the resolver to v1 was done in 5abff3753a7c ("Explicit set workspace.resolver ...") in order to suppress warnings. Since there is no specific reason not to use the new resolver and since it fixes issues, change to `resolver = "2"` everywhere except library.

<!-- homu-ignore:start -->
<!-- homu-ignore:end -->

tgross35 commented:
Split off of https://github.com/rust-lang/rust/pull/128359 since the library was split off to its own `Cargo.lock`.

tgross35 commented on src/tools/tidy/src/deps.rs:
```diff
@@ -98,13 +98,14 @@ const EXCEPTIONS: ExceptionList = &[
-    ("wat", "Apache-2.0 WITH LLVM-exception"),               // rustc
-    ("wit-component", "Apache-2.0 WITH LLVM-exception"),     // rustc
-    ("wit-parser", "Apache-2.0 WITH LLVM-exception"),        // rustc
+    ("wasi-preview1-component-adapter-provider", "Apache-2.0 WITH LLVM-exception"), // rustc
+    // ("wasm-encoder", "Apache-2.0 WITH LLVM-exception"),      // rustc
+    // ("wasm-metadata", "Apache-2.0 WITH LLVM-exception"),     // rustc
+    // ("wasmparser", "Apache-2.0 WITH LLVM-exception"),        // rustc
+    // ("wast", "Apache-2.0 WITH LLVM-exception"),              // rustc
+    // ("wat", "Apache-2.0 WITH LLVM-exception"),               // rustc
+    // ("wit-component", "Apache-2.0 WITH LLVM-exception"),     // rustc
+    // ("wit-parser", "Apache-2.0 WITH LLVM-exception"),        // rustc
```
`wasmparser` changed from `Apache-2.0 WITH LLVM-exception` to `Apache-2.0 WITH LLVM-exception OR Apache-2.0 OR MIT`, but we happen to have both versions (before and after license change) in the lockfile. Regardless of what I do here it seems like tidy complains about one or the other.

This will be resolved if `object` can update their dependency https://github.com/gimli-rs/object/pull/715

tgross35 replied:
There was a release to `object`, this is resolved.

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10262960268/job/28393741763) [(plain)](https://github.com/rust-lang/rust/commit/d9eb9c6f771fdbb7ba9eb1ba6f91233fcc0e9577/checks/28393741763/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.619 Building wheels for collected packages: reuse
#13 2.620   Building wheel for reuse (pyproject.toml): started
#13 2.862   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 2.863   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 2.864   Stored in directory: /tmp/pip-ephem-wheel-cache-wdt_2xbx/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 2.866 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.258 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.258 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.775 Collecting virtualenv
#13 3.775 Collecting virtualenv
#13 3.836   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 3.917      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 72.5 MB/s eta 0:00:00
#13 3.968 Collecting platformdirs<5,>=3.9.1
#13 3.971   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 3.989 Collecting distlib<1,>=0.3.7
#13 3.999      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 91.7 MB/s eta 0:00:00
#13 4.031 Collecting filelock<4,>=3.12.2
#13 4.035   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 4.035   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 4.116 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.302 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 4.4s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      208832 kB
DirectMap2M:     8179712 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/8c7e0e160831866bc1a40691a39455aac21271c0/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-8c7e0e160831866bc1a40691a39455aac21271c0-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
##[endgroup]
fmt check
fmt: checked 5428 files
tidy check
tidy error: invalid license `Apache-2.0 WITH LLVM-exception` in `registry+https://github.com/rust-lang/crates.io-index#wasm-encoder@0.210.0`
tidy error: invalid license `Apache-2.0 WITH LLVM-exception` in `registry+https://github.com/rust-lang/crates.io-index#wasmparser@0.214.0`
tidy error: invalid license `Apache-2.0 WITH LLVM-exception` in `registry+https://github.com/rust-lang/crates.io-index#wasm-encoder@0.210.0`
tidy error: invalid license `Apache-2.0 WITH LLVM-exception` in `registry+https://github.com/rust-lang/crates.io-index#wasmparser@0.214.0`
removing old virtual environment
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (24.1)
  Using cached pip-24.2-py3-none-any.whl.metadata (3.6 kB)
Using cached pip-24.2-py3-none-any.whl (1.8 MB)
Installing collected packages: pip
  Attempting uninstall: pip
---
All checks passed!
checking C++ file formatting
some tidy checks failed

Command LD_LIBRARY_PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-bootstrap-tools/x86_64-unknown-linux-gnu/release/deps:/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/lib" RUSTC="/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/rustc" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/rust-tidy" "/checkout" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "/checkout/obj/build" "4" "--extra-checks=py:lint,cpp:fmt" (failure_mode=DelayFail) did not execute successfully.
Created at: src/core/build_steps/tool.rs:1113:23
Executed at: src/core/build_steps/test.rs:1078:29

Build completed unsuccessfully in 0:01:18
```

</details>

tgross35 committed:
Switch to using the v2 resolver in most workspaces

Pinning the resolver to v1 was done in 5abff3753a7c ("Explicit set
workspace.resolver ...") in order to suppress warnings. Since there is
no specific reason not to use the new resolver and since it fixes
issues, change to `resolver = "2"` everywhere except library and
submodules.

tgross35 committed:
Update some dependency versions that allow better licensing

With the new resolver, a few dependencies get brought in twice with
different licenses. For example, all dependencies from `wasm-tools`
gained Apache-2.0 and MIT options, and with the v2 resolver we were
using one version from before and one version from after this change.
This made tidy's license check difficult.

Update some minimum versions to remove duplicate dependencies and smooth
out license checking.

tgross35 committed:
Run `cargo update` with the new v2 resolver

v2 resolves some dependencies differently, and we adjusted some
dependency versions. Run `cargo update` to make sure everything is in
sync.

tgross35 committed:
Adjust licensing exceptions for WASM components

Recent versions of wasm-tools are now Apache-2.0 or MIT or Apache-2.0
with the LLVM exception, rather than strictly Apache-2.0 with the LLVM
exception. The only component with the exception has moved to a new
dependency `wasi-preview1-component-adapter-provider`.

tgross35 committed:
Generate completions after version updates

Running `cargo update` changed completion output. Regenerate them here.

klensy commented:
Why updating any deps here? There exist separate CI task (https://github.com/rust-lang/rust/pull/127140, which stuck a little). As i understand, `portable-simd` developed in it's own repo (https://github.com/rust-lang/portable-simd), so shouldn't be changed here?

tgross35 commented:
> Why updating any deps here? There exist separate CI task (#127140, which stuck a little). As i understand, `portable-simd` developed in it's own repo (https://github.com/rust-lang/portable-simd), so shouldn't be changed here (at least this way it will be simpler to sync it)?

For licensing - the WASM tools and the crates that depend on them all need to be the latest version. Otherwise we get duplicate versions with conflicting licenses that confuses tidy, as mentioned in https://github.com/rust-lang/rust/pull/128722#discussion_r1705124115. I can rebase after https://github.com/rust-lang/rust/pull/127140 so the diff is smaller.

You are right about portable-simd, I will drop those changes.

bors commented:
:umbrella: The latest upstream changes (presumably #128783) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

Mark-Simulacrum commented:
r=me with this rebased

tgross35 commented:
Thanks Mark.

@bors r=Mark-Simulacrum

bors commented:
:pushpin: Commit 32fbbaa47b146b3f463e28ae15269ec072512cd1 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 32fbbaa47b146b3f463e28ae15269ec072512cd1 -->
<!-- homu: {"type":"Approved","sha":"32fbbaa47b146b3f463e28ae15269ec072512cd1","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

klensy commented:
rollup=never?

tgross35 commented:
Hopefully this shouldn't conflict with anything, maybe just

@bors rollup=iffy

bors commented:
:hourglass: Testing commit 32fbbaa47b146b3f463e28ae15269ec072512cd1 with merge dfe6962c3f7897e995760f5a5dd0e92a549424a4...
<!-- homu: {"type":"BuildStarted","head_sha":"32fbbaa47b146b3f463e28ae15269ec072512cd1","merge_sha":"dfe6962c3f7897e995760f5a5dd0e92a549424a4"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-17`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10443098677/job/28916124289) [(plain)](https://github.com/rust-lang/rust/commit/32fbbaa47b146b3f463e28ae15269ec072512cd1/checks/28916124289/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
------
 > importing cache manifest from ghcr.io/rust-lang/rust-ci-cache:20d3b4d4a2629cbf7865cdbf92fe47512a7c96658c24253a045ff38e8075cd7fb37ca6fcadfa6e6d093333943ad24f6fc4f163ec5b74fd940de9d5bb03eb4d3b:
------
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-17]
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-17', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-17/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---
  Downloaded boml v0.3.1
   Compiling boml v0.3.1
   Compiling y v0.1.0 (/checkout/compiler/rustc_codegen_gcc/build_system)
    Finished `release` profile [optimized] target(s) in 3.65s
     Running `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-codegen/x86_64-unknown-linux-gnu/release/y test --use-system-gcc --use-backend gcc --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_gcc --release --mini-tests --std-tests`
Using system GCC
[BUILD] example
[AOT] mini_core_hello_world
/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_gcc/mini_core_hello_world
abc
---
1 error: unportable markdown
-   --> $DIR/unportable-markdown.rs:21:10
+   --> $DIR/unportable-markdown.rs:49:5
3    |
- LL | /// test [^foo][^bar]
-    |          ^^^^^^
+ LL | /// >bar
6    |
-    = help: confusing footnote reference and link
-    = help: confusing footnote reference and link
+    = help: confusing block quote with no space after the `>` marker
9   --> $DIR/unportable-markdown.rs:8:9
10    |

11 LL | #![deny(rustdoc::unportable_markdown)]
11 LL | #![deny(rustdoc::unportable_markdown)]
12    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
- help: if it should not be a footnote, escape it
-    |
- LL | /// test \[^foo][^bar]
- help: if the footnote is intended, add a space
-    |
-    |
- LL | /// test [^foo] [^bar]
- 
- error: unportable markdown
-   --> $DIR/unportable-markdown.rs:49:5
-    |
-    |
- LL | /// >bar
-    |     ^
-    |
-    = help: confusing block quote with no space after the `>` marker
29 help: if the quote is intended, add a space
30    |
31 LL | /// > bar
35 LL | /// \>bar
36    |     +
37 
- error: aborting due to 2 previous errors
---
To only update this specific test, also pass `--test-args unportable-markdown.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" "/checkout/tests/rustdoc-ui/unportable-markdown.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/unportable-markdown" "-A" "internal_features" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/unportable-markdown/auxiliary"
--- stderr -------------------------------
error: unportable markdown
##[error]  --> /checkout/tests/rustdoc-ui/unportable-markdown.rs:49:5
   |
   |
LL | /// >bar
   |     ^
   |
   = help: confusing block quote with no space after the `>` marker
  --> /checkout/tests/rustdoc-ui/unportable-markdown.rs:8:9
   |
LL | #![deny(rustdoc::unportable_markdown)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: if the quote is intended, add a space
   |
LL | /// > bar
help: if it should not be a quote, escape it
   |
LL | /// \>bar
   |     +
```

</details>

rust-log-analyzer commented:
The job **`aarch64-apple`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/10443306396/job/28916562490) [(plain)](https://github.com/rust-lang-ci/rust/commit/dfe6962c3f7897e995760f5a5dd0e92a549424a4/checks/28916562490/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
1 error: unportable markdown
-   --> $DIR/unportable-markdown.rs:21:10
+   --> $DIR/unportable-markdown.rs:49:5
3    |
- LL | /// test [^foo][^bar]
-    |          ^^^^^^
+ LL | /// >bar
6    |
-    = help: confusing footnote reference and link
-    = help: confusing footnote reference and link
+    = help: confusing block quote with no space after the `>` marker
9   --> $DIR/unportable-markdown.rs:8:9
10    |

11 LL | #![deny(rustdoc::unportable_markdown)]
11 LL | #![deny(rustdoc::unportable_markdown)]
12    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
- help: if it should not be a footnote, escape it
-    |
- LL | /// test \[^foo][^bar]
- help: if the footnote is intended, add a space
-    |
-    |
- LL | /// test [^foo] [^bar]
- 
- error: unportable markdown
-   --> $DIR/unportable-markdown.rs:49:5
-    |
-    |
- LL | /// >bar
-    |     ^
-    |
-    = help: confusing block quote with no space after the `>` marker
29 help: if the quote is intended, add a space
30    |
31 LL | /// > bar
35 LL | /// \>bar
36    |     +
37 
- error: aborting due to 2 previous errors
---
To only update this specific test, also pass `--test-args unportable-markdown.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/stage2/bin/rustdoc" "/Users/runner/work/rust/rust/tests/rustdoc-ui/unportable-markdown.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/work/rust/rust/vendor" "--sysroot" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/stage2" "--target=aarch64-apple-darwin" "--check-cfg" "cfg(FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-o" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/test/rustdoc-ui/unportable-markdown" "-A" "internal_features" "-Cdebuginfo=0" "-Lnative=/Users/runner/work/rust/rust/build/aarch64-apple-darwin/native/rust-test-helpers" "-L" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/test/rustdoc-ui/unportable-markdown/auxiliary"
--- stderr -------------------------------
error: unportable markdown
##[error]  --> /Users/runner/work/rust/rust/tests/rustdoc-ui/unportable-markdown.rs:49:5
   |
   |
LL | /// >bar
   |     ^
   |
   = help: confusing block quote with no space after the `>` marker
  --> /Users/runner/work/rust/rust/tests/rustdoc-ui/unportable-markdown.rs:8:9
   |
LL | #![deny(rustdoc::unportable_markdown)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: if the quote is intended, add a space
   |
LL | /// > bar
help: if it should not be a quote, escape it
   |
LL | /// \>bar
   |     +
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/10443306396/job/28917287405)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/10443306396/job/28917287405","builder_name":"checks-actions"} -->

tgross35 commented:
@rust-lang/rustdoc something changed between pulldown-cmark 0.11.0 and 0.11.2 used by `rustc_resolve` that apparently makes it stop emitting the `unportable_markdown` lint `confusing footnote reference and link` https://github.com/rust-lang/rust/actions/runs/10443098677/job/28916124289?pr=128722#step:25:6880. Would you prefer me to remove that error annotation for now, or pin pulldown-cmark to 0.11.0? Or is there an easy fix here?

Relevant test: https://github.com/rust-lang/rust/blob/6de928dce9472b864f4e5d590dd7aa2075cb7551/tests/rustdoc-ui/unportable-markdown.rs#L21-L22

tgross35 committed:
Adjust expected errors for a `rustdoc` test

`pulldown-cmark` has slightly different behavior between 0.11.0 and
0.11.2, causing one of the `unportable-markdown` tests to no longer emit
an error. Per [1], remove the error annotation and bless the output.

[1]: https://github.com/rust-lang/rust/pull/128722#issuecomment-2295522292

tgross35 commented:
Thanks, I updated this test in 1eb1e1816d3e42f3f31be75aae3c147302ae02a6.

@bors r=Mark-Simulacrum

bors commented:
:pushpin: Commit 370168787bd6f2ed26788a51af97166d868e0609 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 370168787bd6f2ed26788a51af97166d868e0609 -->
<!-- homu: {"type":"Approved","sha":"370168787bd6f2ed26788a51af97166d868e0609","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 370168787bd6f2ed26788a51af97166d868e0609 with merge 05c407741a803fa07e8dd8ddff4bbf98e1acd183...
<!-- homu: {"type":"BuildStarted","head_sha":"370168787bd6f2ed26788a51af97166d868e0609","merge_sha":"05c407741a803fa07e8dd8ddff4bbf98e1acd183"} -->

rust-log-analyzer commented:
The job **`x86_64-msvc-ext`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/10447261576/job/28925924246) [(plain)](https://github.com/rust-lang-ci/rust/commit/05c407741a803fa07e8dd8ddff4bbf98e1acd183/checks/28925924246/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] miri test:false 4.514
error: failed to remove file `C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage1-tools\x86_64-pc-windows-msvc\release\miri.exe`

Caused by:
  Access is denied. (os error 5)
Command has failed. Rerun with -v to see more details.
  local time: Mon, Aug 19, 2024  6:21:46 AM
  network time: Mon, 19 Aug 2024 06:21:46 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/10447261576/job/28928457668)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/10447261576/job/28928457668","builder_name":"checks-actions"} -->

tgross35 commented:
Looks like one of the spurious Windows failures.

@bors retry

bors commented:
:hourglass: Testing commit 370168787bd6f2ed26788a51af97166d868e0609 with merge 45fbf41deb24581471e3e56824d9318d3d415cb8...
<!-- homu: {"type":"BuildStarted","head_sha":"370168787bd6f2ed26788a51af97166d868e0609","merge_sha":"45fbf41deb24581471e3e56824d9318d3d415cb8"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/10451128333/job/28942938218)
Approved by: Mark-Simulacrum
Pushing 45fbf41deb24581471e3e56824d9318d3d415cb8 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/10451128333/job/28942938218"},"merge_sha":"45fbf41deb24581471e3e56824d9318d3d415cb8"} -->

rust-timer commented:
Finished benchmarking commit (45fbf41deb24581471e3e56824d9318d3d415cb8): [comparison URL](https://perf.rust-lang.org/compare.html?start=e3f909b2bbd0b10db6f164d466db237c582d3045&end=45fbf41deb24581471e3e56824d9318d3d415cb8&stat=instructions:u).

### Overall result: ❌ regressions - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=e3f909b2bbd0b10db6f164d466db237c582d3045&end=45fbf41deb24581471e3e56824d9318d3d415cb8&stat=instructions:u)
This is a highly reliable metric that was used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 1.0% | [1.0%, 1.0%] | 1     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=e3f909b2bbd0b10db6f164d466db237c582d3045&end=45fbf41deb24581471e3e56824d9318d3d415cb8&stat=max-rss)
<details>
<summary>Results (primary 2.2%, secondary -0.3%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | 2.2%  | [2.2%, 2.2%]   | 1     |
| Regressions ❌ <br /> (secondary)  | 1.4%  | [1.4%, 1.4%]   | 2     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -2.0% | [-2.0%, -2.0%] | 2     |
| All ❌✅ (primary)                 | 2.2%  | [2.2%, 2.2%]   | 1     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=e3f909b2bbd0b10db6f164d466db237c582d3045&end=45fbf41deb24581471e3e56824d9318d3d415cb8&stat=cycles:u)
<details>
<summary>Results (primary -4.5%, secondary -2.5%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -4.5% | [-4.5%, -4.5%] | 1     |
| Improvements ✅ <br /> (secondary) | -2.5% | [-2.5%, -2.5%] | 1     |
| All ❌✅ (primary)                 | -4.5% | [-4.5%, -4.5%] | 1     |
</details>

### [Binary size](https://perf.rust-lang.org/compare.html?start=e3f909b2bbd0b10db6f164d466db237c582d3045&end=45fbf41deb24581471e3e56824d9318d3d415cb8&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 750.46s -> 749.421s (-0.14%)
**Artifact size**: 339.15 MiB -> 336.74 MiB (-0.71%)

dpaoliello commented:
@rustbot label +relnotes

This change includes a [new version of cc-rs which no longer supports Visual Studio 2013](https://github.com/rust-lang/cc-rs/pull/1046), which means that Rust will no longer use or consider the VS 2013 linker when building `*-msvc` targets.

Mark-Simulacrum commented:
@tgross35 can we back that cc change out? It shouldn't go through a PR like this - probably merits t-compiler FCP?

(I'm not sure if we document which linker versions we consider, but I'm not comfortable landing it in a PR like this).

dpaoliello commented:
Per INSTALL.md, VS 2017 is required: https://github.com/rust-lang/rust/blob/636d7ff91b9847d6d43c7bbe023568828f6e3246/INSTALL.md?plain=1#L199

Rustup's docs have also been updated, but it looks like it hasn't been deployed yet: https://github.com/rust-lang/rustup/commit/48317011e6a41aadf0e9e822307840f0df44298a

That said, I don't think there's anything **blocking** older versions except for cc-rs support.

Mark-Simulacrum commented:
INSTALL.md is ... not really the right place for that to be documented, it should probably go on the platform support page (rendered https://doc.rust-lang.org/nightly/rustc/platform-support.html) and doesn't appear there currently. I don't have any objection to raising the limit, but we should do that intentionally.

tgross35 commented:
I opened https://github.com/rust-lang/rust/pull/129290 to pin to the version below that release, per https://github.com/rust-lang/cc-rs/pull/1142. Bit surprising that this change was semver-compatible, but maybe that wasn't explicitly intentional.

@dpaoliello since you have the background here, maybe you could put up a PR upgrading to 1.1.x and changing the docs after my pinning PR merges? As a place for compiler FCP to get us unpinned.

dpaoliello commented:
> changing the docs 

I assume you mean create a platform support page for the `-msvc` targets?

> Bit surprising that this change was semver-compatible, but maybe that wasn't explicitly intentional.

If I recall, we went ahead with the change since 1) VS 2013 is out of support and 2) cc-rs's MSRV was bumped past the version where the Rust's INSTALL.md indicated that VS2017 was required.

Seems like there should be some guidance for bumping and documenting min tool versions somewhere. I know we've been waiting on changing the raw-dylib feature since we need a newer binutils version, but I can't find any docs that indicate what the min required version is.

Mark-Simulacrum commented:
Removing relnotes since https://github.com/rust-lang/rust/pull/129290 dropped the break.

Veykril commented:
This seems to have broken most (if not all) `./x.py test compiler/<insert crate name here>` invocations, failing with 
```
error: none of the selected packages contains these features: jemalloc, llvm, max_level_info, rustc_use_parallel_compiler
```
for `compiler/rustc_abi` for example

tgross35 commented:
Maybe bootstrap needs to get adjusted to be more accurate with the feature flags it passes? Might be worth a new issue
